x/auth/vesting/msg_server.go

```diff
@@ -57,28 +52,18 @@ func (s msgServer) CreateVestingAccount(goCtx context.Context, msg *types.MsgCre
 		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "account %s already exists", msg.ToAddress)
 	}
 
-	account := ak.NewAccountWithAddress(ctx, to)
-	baseAccount, ok := account.(*authtypes.BaseAccount)
-	if !ok {
-		if getter, ok := account.(baseAccountGetter); ok {
-			baseAccount = getter.GetBaseAccount()
-		}
-	}
-	if baseAccount == nil {
-		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "invalid account type; expected: BaseAccount, got: %T", baseAccount)
-	}
-
+	baseAccount := authtypes.NewBaseAccountWithAddress(to)
+	baseAccount = ak.NewAccount(ctx, baseAccount).(*authtypes.BaseAccount)
```

Comment:
Are we ok with not checking if the type cast went ok?

Reply:
We know that `NewBaseAccountWithAddress` will return a `&BaseAccount`. I suppose we could add an explicit check :)

I will do that

Reply:
There are a couple more of these, so let's add it to all of them for good measure

Reply:
Actually there is no need to type check. It just creates more boilerplate IMO. We create a concrete BaseAccount and then we pass it to `NewAccount`, which just returns what we pass in.